zerver/actions/message_send.py

```diff
@@ -1251,15 +1251,15 @@ def send_pm_if_empty_stream(
     if not sender.is_bot or sender.bot_owner is None:
         return
 
-    arg_dict = {
-        "bot_identity": f"`{sender.delivery_email}`",
-        "stream_id": stream_id,
-        "stream_name": f"#**{stream_name}**",
-        "new_stream_link": "#streams/new",
-    }
     if sender.bot_owner is not None:
         with override_language(sender.bot_owner.default_language):
             if stream is None:
+                arg_dict = {
+                    "bot_identity": f"`{sender.delivery_email}`",
+                    "stream_id": stream_id,
+                    "stream_name": f"#**{stream_name}**",
+                    "new_stream_link": "#streams/new",
+                }
```

Comment:
Are we assuming here that in this `if stream is None` case, `stream_name` is supposed to not be `None`, ensured by the caller? Do we want an assert above this if so?

Reply:
Yeah an assert would be reasonable here.

Reply:
Actually a better idea might be to just do the `arg_dict` inside the `if/else` blocks here, so we never put a `None` into an `arg_dict`?

Reply:
Sure.  Repushed.